src/cpu/pred/BranchPredictor.py

```diff
@@ -100,6 +100,12 @@ class SimpleIndirectPredictor(IndirectPredictor):
     indirectPathLength = Param.Unsigned(
         3, "Previous indirect targets to use for path history"
     )
+    speculativePathLength = Param.Unsigned(
```

Comment:
How does this interact with the O3 CPU? Would it make sense to make this unbounded instead (i.e., assuming it is sized to match the speculative capabilities of the CPU)?

Reply:
Before, there was no speculative path length only normal path length.
I encountered that this could cause corruption if the CPU speculates too far ahead.
We could set this parameter to a high value by default to make it "unbounded"